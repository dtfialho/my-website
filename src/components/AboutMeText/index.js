import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import * as S from './styled'

const AboutMeText = () => {
  const { me } = useStaticQuery(
    graphql`
      query {
        me: file(relativePath: { eq: "me.jpeg" }) {
          childImageSharp {
            fixed(width: 200, height: 200) {
              ...GatsbyImageSharpFixed
            }
          }
        }
      }
    `
  )

  return (
    <S.AboutMeTextContainer>
      <S.AboutMeFigure>
        <S.AboutMeImage fixed={me.childImageSharp.fixed} />
      </S.AboutMeFigure>
      
      <S.AboutMeTextParagraph>
        Olá! Meu nome é Diego Teixeira Fialho, nasci em Petrópolis/RJ e atualmente trabalho como Frontend Developer na <a href="http://www.personare.com.br" target="_blank" rel="noopener noreferrer">Personare</a>. Já trabalhei como Fullstack Developer na <a href="https://www.maxisite.net/" target="_blank" rel="noopener noreferrer">Maxisite</a> e Frontend Developer na <a href="https://www.alterdata.com.br/" target="_blank" rel="noopener noreferrer">Alterdata Software</a>.
      </S.AboutMeTextParagraph>

      <S.AboutMeTextParagraph>
        Apaixonado por tecnologia desde cedo sempre fui curioso para aprender. Aos 20 anos, fiz meu primeiro curso na área de tecnologia no SENAI e aos 23 ingressei na faculdade de Tecnologia da Informação e Comunicação pela FAETERJ - Petrópolis.
      </S.AboutMeTextParagraph>

      <S.AboutMeTextParagraph>
        Amo desenvolver e estudar novas tecnologias que surgem a cada dia para facilitar a nossa vida como desenvolvedores. Com o sangue metade café e metade código (e um pouco de World of Warcraft que é o meu passatempo favorito btw), amo criar e idealizar componentes utilizando frameworks Javascript e conceitos para um código limpo e bem estruturado.
      </S.AboutMeTextParagraph>
    </S.AboutMeTextContainer>
  )
}

export default AboutMeText
