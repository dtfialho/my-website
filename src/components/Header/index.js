import React from 'react'
import PropTypes from 'prop-types'
import Logo from '../Logo'
import Navbar from '../Navbar'
import * as S from './styled'

const Header = ({ isFixed }) => (
  <S.HeaderElement isFixed={isFixed}>
    <S.HeaderLogoLink to="/">
      <Logo />
    </S.HeaderLogoLink>
    <Navbar />
  </S.HeaderElement>
)

Header.propTypes = {
  isFixed: PropTypes.bool
}

export default Header
