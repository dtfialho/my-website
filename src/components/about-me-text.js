import React from 'react'
import me from '../images/me.jpeg'

const AboutMeText = () => (
  <div className="about-me__text">
    <figure className="me">
      <img src={me} alt="Diego T. Fialho" />
    </figure>
    
    <p>
      Olá! Meu nome é Diego Teixeira Fialho, nasci em Petrópolis/RJ e atualmente trabalho como Frontend Developer na <a href="http://www.personare.com.br" target="_blank" rel="noopener noreferrer">Personare</a>. Já trabalhei como Fullstack Developer na <a href="https://www.maxisite.net/" target="_blank" rel="noopener noreferrer">Maxisite</a> e Frontend Developer na <a href="https://www.alterdata.com.br/" target="_blank" rel="noopener noreferrer">Alterdata Software</a>.
    </p>

    <p>
      Apaixonado por tecnologia desde cedo sempre fui curioso para aprender. Aos 20 anos, fiz meu primeiro curso na área de tecnologia no SENAI e aos 23 ingressei na faculdade de Tecnologia da Informação e Comunicação pela FAETERJ - Petrópolis.
    </p>

    <p>
      Amo desenvolver e estudar novas tecnologias que surgem a cada dia para facilitar a nossa vida como desenvolvedores. Com o sangue metade café e metade código (e um pouco de World of Warcraft que é o meu passatempo favorito btw), amo criar e idealizar componentes utilizando frameworks Javascript e conceitos para um código limpo e bem estruturado.
    </p>
  </div>
)

export default AboutMeText
