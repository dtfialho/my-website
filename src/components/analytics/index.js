import Script from 'next/script'
import { Analytics as VercelAnalytics } from '@vercel/analytics/react'

const Analytics = () => {
  const GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_TRACKING

  return (
    <>
      <VercelAnalytics />
      <Script
        strategy="afterInteractive"
        src={`https://www.googletagmanager.com/gtag/js?id=${GA_TRACKING_ID}`}
      />
      <Script
        id="google-analytics"
        strategy="afterInteractive"
        dangerouslySetInnerHTML={{
          __html: `
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
        
            gtag('config', '${GA_TRACKING_ID}');
          `
        }}
      />
    </>
  )
}

export default Analytics
