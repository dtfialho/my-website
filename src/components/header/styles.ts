import styled, { css } from 'styled-components'

const wrapperModifiers = {
  fixed: () => css`
    position: fixed;
    top: 0;
    left: 0;
    z-index: 2;
  `
}

interface WrapperProps {
  fixed: boolean
}

export const Wrapper = styled.header<WrapperProps>`
  ${({ fixed }) => css`
    width: 100vw;
    padding: 20px 40px;
    display: flex;
    flex-flow: row;
    justify-content: space-between;
    align-items: center;

    ${fixed && wrapperModifiers.fixed()}
  `}
`

export const LogoLink = styled.span`
  svg {
    width: 50px;
  }
`
