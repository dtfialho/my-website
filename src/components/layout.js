import React from 'react'
import PropTypes from 'prop-types'
import GlobalStyles from '../styles/global'
import SEO from './seo'
import Navbar from './navbar'

const Layout = ({ title, children }) => (
  <main>
    <GlobalStyles />
    <SEO title={title} />
    <header>
      <Navbar />
    </header>

    {children}
  </main>
)

Layout.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
}

export default Layout
