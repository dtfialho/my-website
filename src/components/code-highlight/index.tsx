import { LightAsync as SyntaxHighlighter } from 'react-syntax-highlighter'
import { dracula } from 'react-syntax-highlighter/dist/cjs/styles/hljs'

type CodeHighlightProps = {
  language?: string
  value: string
}

const CodeHighlight = ({ language = '', value }: CodeHighlightProps) => (
  <SyntaxHighlighter
    language={language}
    style={dracula}
    customStyle={{ paddingLeft: '16px', paddingRight: '16px' }}
  >
    {value}
  </SyntaxHighlighter>
)

export default CodeHighlight
