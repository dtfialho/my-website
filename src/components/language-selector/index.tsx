import Image from 'next/image'
import { useRouter } from 'next/router'
import { useContext } from 'react'

import { LanguageSelectorContext } from './provider'
import * as S from './styles'

const LanguageSelector = () => {
  const { setShowModal } = useContext(LanguageSelectorContext)
  const { locale: activeLocale } = useRouter()

  const handleOpenModal = () => {
    setShowModal(true)
  }

  return (
    <>
      <S.Button type="button" onClick={handleOpenModal}>
        <Image
          src={`/img/${activeLocale}.jpg`}
          width="29"
          height="22"
          alt="Active language"
        />
      </S.Button>
    </>
  )
}

export default LanguageSelector
