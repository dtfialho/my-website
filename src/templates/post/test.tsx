import format from 'date-fns/format'
import { screen, render } from '@testing-library/react'

import * as Header from 'components/header'
import Post, { PostProps } from './'

jest.mock('remark-gfm', () => ({}))
jest.mock('rehype-raw', () => ({}))

jest.mock('components/header')
const mockedHeader = Header.default as jest.Mock
mockedHeader.mockImplementation(() => <header>Header</header>)

const postProps: PostProps = {
  content:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Sapiente fugit numquam tempora voluptatum voluptas in, doloremque maiores natus cum voluptate culpa perferendis ipsam pariatur unde optio id amet dolorum inventore.',
  title: 'Luke is comming! :D',
  date: '2022-06-29T00:00:00-03:00',
  hero_image: '/img/bg.jpg'
}

describe('Templates/Post', () => {
  it('Should render correctly', () => {
    const { container } = render(<Post {...postProps} />)
    expect(container).toMatchSnapshot()
  })

  it('Should render the passed props correctly', async () => {
    const parsedDate = format(new Date(postProps.date), 'dd/MM/yyyy')
    const dateRegex = new RegExp(parsedDate)

    render(<Post {...postProps} />)
    expect(await screen.findByRole('heading', { name: postProps.title }))
    expect(screen.queryByText(dateRegex)).toBeInTheDocument()
    expect(screen.getByText(postProps.content)).toBeInTheDocument()
  })
})
