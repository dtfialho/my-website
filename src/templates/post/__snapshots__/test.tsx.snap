// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/Post Should render correctly 1`] = `
.c0 {
  max-width: 912px;
  margin: 0 auto;
  padding: 153px 16px 100px;
}

.c1 a {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 pre,
.c1 ul,
.c1 iframe {
  margin-bottom: 24px;
}

.c1 p code,
.c1 li code {
  display: inline-block;
  padding: 2px 4px;
  color: #c7254e;
  background-color: #f9f2f4;
  border-radius: 4px;
}

.c1 pre {
  line-height: 1.4;
  color: #fff;
  background-color: #282c34;
  border-radius: 6px;
  cursor: default;
  max-width: none;
}

.c2 {
  text-align: center;
  margin-bottom: 0;
}

.c3 {
  display: block;
  text-align: center;
  margin-bottom: 32px;
}

.c4 {
  position: relative;
  height: 250px;
  margin-bottom: 32px;
}

.c5 {
  background-color: #4c57ba;
  color: #fff;
  border-radius: 6px;
  padding: 8px 24px;
  margin-top: 32px;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c5:hover {
  opacity: 0.8;
}

@media (min-width:768px) {
  .c0 {
    padding-top: 193px;
  }
}

@media (min-width:768px) {
  .c3 {
    margin-bottom: 40px;
  }
}

@media (min-width:768px) {
  .c4 {
    height: 300px;
    margin-bottom: 40px;
  }
}

<div>
  <header>
    Header
  </header>
  <main
    class="c0"
  >
    <article
      class="c1"
    >
      <h1
        class="c2"
      >
        Luke is comming! :D
      </h1>
      <p
        class="c3"
      >
        <small>
          Postado em
          : 
          29/06/2022
        </small>
      </p>
      <div
        class="c4"
      >
        <img
          alt="Luke is comming! :D"
          layout="fill"
          objectfit="contain"
          src="/img/bg.jpg"
        />
      </div>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Sapiente fugit numquam tempora voluptatum voluptas in, doloremque maiores natus cum voluptate culpa perferendis ipsam pariatur unde optio id amet dolorum inventore.
      </p>
    </article>
    <a
      class="c5"
      href="/blog"
    >
      Voltar
    </a>
  </main>
</div>
`;

exports[`Templates/Post Should render correctly in en 1`] = `
.c0 {
  max-width: 912px;
  margin: 0 auto;
  padding: 153px 16px 100px;
}

.c1 a {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 pre,
.c1 ul,
.c1 iframe {
  margin-bottom: 24px;
}

.c1 p code,
.c1 li code {
  display: inline-block;
  padding: 2px 4px;
  color: #c7254e;
  background-color: #f9f2f4;
  border-radius: 4px;
}

.c1 pre {
  line-height: 1.4;
  color: #fff;
  background-color: #282c34;
  border-radius: 6px;
  cursor: default;
  max-width: none;
}

.c2 {
  text-align: center;
  margin-bottom: 0;
}

.c3 {
  display: block;
  text-align: center;
  margin-bottom: 32px;
}

.c4 {
  position: relative;
  height: 250px;
  margin-bottom: 32px;
}

.c5 {
  background-color: #4c57ba;
  color: #fff;
  border-radius: 6px;
  padding: 8px 24px;
  margin-top: 32px;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c5:hover {
  opacity: 0.8;
}

@media (min-width:768px) {
  .c0 {
    padding-top: 193px;
  }
}

@media (min-width:768px) {
  .c3 {
    margin-bottom: 40px;
  }
}

@media (min-width:768px) {
  .c4 {
    height: 300px;
    margin-bottom: 40px;
  }
}

<div>
  <header>
    Header
  </header>
  <main
    class="c0"
  >
    <article
      class="c1"
    >
      <h1
        class="c2"
      >
        Luke is comming! :D
      </h1>
      <p
        class="c3"
      >
        <small>
          Posted on
          : 
          29/06/2022
        </small>
      </p>
      <div
        class="c4"
      >
        <img
          alt="Luke is comming! :D"
          layout="fill"
          objectfit="contain"
          src="/img/bg.jpg"
        />
      </div>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Sapiente fugit numquam tempora voluptatum voluptas in, doloremque maiores natus cum voluptate culpa perferendis ipsam pariatur unde optio id amet dolorum inventore.
      </p>
    </article>
    <a
      class="c5"
      href="/blog"
    >
      Back
    </a>
  </main>
</div>
`;
